hackathons/usoc23: Add USoC'23 landing page and Registration Challenges (#238)
unikraft/docs: content/en/community/hackathons/usoc23/_index.md

---
title: "Unikraft Summer of Code 2023"
date: 2023-04-12T01:10:00+02:00
draft: false
collapsible: true
description: Unikraft Summer of Code 2023, July 4 - 15, 2023
---

## Unikraft Summer of Code 2023

Unikraft Summer of Code 2023 (*USoC'23*) is a **free and virtual** unikernel and library operating systems workshop held by members of the Unikraft community.

The two-week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.

The workshop will be hands-on and will take place for six days, between July 4 and July 15, 2023, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
Topics include building unikernels, benchmarking, debugging, porting applications, virtualization and platform specifics.

> The workshops will be followed by a full day hackathon which will be held live in `TBA` at [University POLITEHNICA of Bucharest](https://upb.ro), with the possibility to also join online via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).

Reviewer: Update TBA

Author: Should it be room PR706?

Reviewer: Mention it will be online (this is the default). For those how can make it, we will also organize the hackathon in room PR706.

---
title: "Unikraft Summer of Code 2023"
date: 2023-04-12T01:10:00+02:00
draft: false
collapsible: true

> description: Unikraft Summer of Code 2023, July 3 - 15, 2023

Reviewer 2: ```suggestion
description: Unikraft Summer of Code 2023, July 4 - 15, 2023
```

> image: /assets/imgs/usoc23.png

Reviewer: Remove this image, until we have a poster ready.

---

## Unikraft Summer of Code 2023

Unikraft Summer of Code 2023 (*USoC'23*) is a **free and virtual** unikernel and library operating systems workshop held by members of the Unikraft community.

> From zero-to-hero the two week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.

Reviewer: ```suggestion
The two week event holds a number of starter tutorials, workshops and lessons on how to get build Unikraft unikernels for production systems.
```

> The workshop will be hands-on and will take place for 10 days, between July 4 and July 15, 2023, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).

Reviewer: ```suggestion
The workshop will be hands-on and will take place for six days, between July 4 and July 15, 2023, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

> The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-8pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).

Reviewer 3: ```suggestion
The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

> The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).

Reviewer 2: ```suggestion
The workshop will be hands-on and will take place for 10 days, between August 29 and September 9, 2022, 4pm-7pm CEST, in English, virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

Maybe put TODO on the dates for now?

Author: We know the dates, I just missed them here.

Topics include building unikernels, benchmarking, debugging, porting applications, virtualization and platform specifics.

> The workshop will take place for two weeks (July 04 - July 16, 4pm-7pm CEST) virtually via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).

Reviewer: This line needs to go away, it's already mentioned above.

> There will be 6 3hrs sessions, followed by a full day hackathon will be held live at on `TBA` at [University POLITEHNICA of Bucharest](https://upb.ro), with the possibility to also join online via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).

Reviewer 3: ```suggestion
There will be six 3-hour sessions, followed by a full day hackathon which will be held live at in `TBA` at [University POLITEHNICA of Bucharest](https://upb.ro), with the possibility to also join online via [Unikraft's Discord server](https://bit.ly/UnikraftDiscord).
```

### Registration

If you're eager to learn more about operating systems and unikernels, to work on practical open source tasks, to expand your knowledge of low-level topics, you'll want to be part of USoC'23.
To take part in the event, please fill this [registration form](https://forms.gle/ZUVLnkb6YNWhr9Cp8).
You will need to complete a set of challenges that will get you accustomed to the environment you will be using during the sessions.

> It's recommended you checked these prerequisites before taking part in USoC'23:

Reviewer 3: ```suggestion
It's recommended you check these prerequisites before taking part in USoC'23:
```

> * fair knowledge of Linux command-line interface

Reviewer 2: ```suggestion
* good knowledge of Linux command-line interface
```

Author: I would leave this to `fair`, Linux cli is not that strong of an item within Unikraft.

> * fair knowledge of programming concepts and the C programming language

Reviewer 2: ```suggestion
* good knowledge of programming concepts and the C programming language
```

> * basic understanding of operating system concepts: process, threads, virtual memory, filesystems, file descriptors

Reviewer 3: ```suggestion
* basic understanding of operating system concepts: processes, threads, virtual memory, filesystems, file descriptors
```

> * some exposure to assembly language and computer architecture

Reviewer 3: ```suggestion
* some exposure to the assembly language and computer architecture
```

* fondness for software engineering, hacking, tinkering with software components

### People

> USoC'23 will be held by members of the Unikraft community including professors, lecturers, PhD and MSci students from [University POLITEHNICA of Bucharest](https://upb.ro/) and the comercial side of Unikraft, [`Unikraft.io`](https://unikraft.io/).

Reviewer: ```suggestion
USoC'23 will be held by members of the Unikraft community including professors and students from [University POLITEHNICA of Bucharest](https://upb.ro/) and the comercial side of Unikraft, [`Unikraft.io`](https://unikraft.io/).
```

> USoC'23 will be held by members of the Unikraft community including professors, lecturers, PhD and MSci students from [University POLITEHNICA of Bucharest](https://upb.ro/) and the comercial side of Unikraft, [`Unikraft.io`](https://unikraft.io/).

Reviewer 3: Maybe there will be more universities? Will need to update after the list of volunteers is done

Author: Yes, we will update this on the go if it is the case.

Other members of the Unikraft community will provide online support on [Discord](https://bit.ly/UnikraftDiscord).

### Schedule

> USoC'23 consists of 6 sessions in 6 days and a hackathon.

Reviewer 2: ```suggestion
USoC'23 consists of 6 sessions and a hackathon.
```

Each session is 3 hours long and consists of practical demos and then exercises for participants.

> The hackathon is a full day event where you'll be provided tasks and challenges that require you to enable, test, fix, evaluate and improve applications and libraries with Unikraft.

Reviewer 3: Also specify how many hours `10 hours` I think

Author: I will put 10hrs now for reference, and will update it after we decide the exact date, location and any other details.

> The hackathon is a full day event (10 hours) where you'll be provided tasks and challenges that require you to enable, test, fix, evaluate and improve applications and libraries with Unikraft.

Reviewer 2: ```suggestion
The hackathon is a full day event (10 hours) where you'll be provided various tasks and challenges that require you to enable, test, fix, evaluate and improve applications and libraries with Unikraft.
```
